src/PostgREST/Plan.hs

```diff
@@ -376,15 +377,24 @@ initReadRequest ctx@ResolverContext{qi=QualifiedIdentifier{..}} =
 addAliases :: ReadPlanTree -> Either ApiRequestError ReadPlanTree
 addAliases = Right . fmap addAliasToPlan
   where
-    addAliasToPlan rp@ReadPlan{select=sel} = rp{select=map aliasSelectField sel}
+    addAliasToPlan rp@ReadPlan{select=sel, relIsSpread=spr} = rp{select=map (aliasSelectField spr) sel}
 
-    aliasSelectField :: CoercibleSelectField -> CoercibleSelectField
-    aliasSelectField field@CoercibleSelectField{csField=fieldDetails, csAggFunction=aggFun, csAlias=alias}
-      | isJust alias || isJust aggFun = field
+    aliasSelectField :: Bool -> CoercibleSelectField -> CoercibleSelectField
+    aliasSelectField isSpread field@CoercibleSelectField{csField=fieldDetails, csAggFunction=aggFun, csAlias=alias}
+      | isJust alias = field
+      | isJust aggFun = fieldAliasForSpreadAgg isSpread field
       | isJsonKeyPath fieldDetails, Just key <- lastJsonKey fieldDetails = field { csAlias = Just key }
       | isTransformPath fieldDetails = field { csAlias = Just (cfName fieldDetails) }
       | otherwise = field
 
+    fieldAliasForSpreadAgg isSpread field@CoercibleSelectField{csField=fieldDetails, csAggFunction=aggFun} =
+      -- A request like: `/top_table?select=...middle_table(...nested_table(count()))` will `SELECT` the full row instead of `*`,
+      -- because doing a `COUNT(*)` in `top_table` would not return the desired results.
+      -- So we use the "count" alias if none is present since the field name won't be selected.
```

Comment:
I used "count" as an alias because PostgreSQL also returns that name when no alias is given for `SELECT COUNT(...)`. This would conflict when a table has a `count` column or when selecting a `count()` in each nested resource, so the solution would be to use aliases. This is not a breaking change since those requests are not working right now in v12.2.3:

```bash
curl 'localhost:3000/process_supervisor?select=...processes(count,...process_costs(count()))'
curl 'localhost:3000/process_supervisor?select=...processes(count(),...process_costs(count()))'
```
```json
{"code":"42803","details":null,"hint":null,"message":"aggregate functions are not allowed in GROUP BY"}
```

With these changes, the error is much nicer now:

```bash
curl 'localhost:3000/process_supervisor?select=...processes(count,...process_costs(count()))'
```
```json
{"code":"42702","details":null,"hint":null,"message":"column reference \"count\" is ambiguous"}
```